pkg/prom/agent.go

```diff
@@ -62,34 +70,25 @@ func (c *Config) ApplyDefaults() error {
 
 	usedNames := map[string]struct{}{}
 
-	for i, cfg := range c.Configs {
-		err := c.Configs[i].ApplyDefaults(&c.Global)
-		if err != nil {
+	for i := range c.Configs {
+		name := c.Configs[i].Name
+		if err := c.Configs[i].ApplyDefaults(&c.Global); err != nil {
 			// Try to show a helpful name in the error
-			name := cfg.Name
 			if name == "" {
 				name = fmt.Sprintf("at index %d", i)
 			}
 
 			return fmt.Errorf("error validating instance %s: %w", name, err)
 		}
 
-		if _, ok := usedNames[cfg.Name]; ok {
+		if _, ok := usedNames[name]; ok {
 			return fmt.Errorf(
 				"prometheus instance names must be unique. found multiple instances with name %s",
-				cfg.Name,
+				name,
 			)
 		}
-		usedNames[cfg.Name] = struct{}{}
+		usedNames[name] = struct{}{}
 	}
-
-	for i := range c.Configs {
-		err := c.Configs[i].ApplyDefaults(&c.Global)
-		if err != nil {
-			return err
-		}
-	}
```

Comment:
Thanks for taking this out! I'm not sure how this ended up in here twice but you're right - this was redundant.